FoodFinder/ViewModels/MapViewModel.swift

```diff
@@ -10,78 +10,21 @@ import Foundation
 import MapKit
 
 protocol MapViewModelDelegate: class {
-    func shouldSetRegion(_ region: MKCoordinateRegion)
-    func shouldAddAnnotations(_ annotations: [MKPointAnnotation])
-//    func didUpdateLocations(_ location: [CLLocation])
     func didFetchRestaurants(_ restaurants: [Restaurant])
 }
 
 class MapViewModel: NSObject {
     weak var delegate: MapViewModelDelegate?
+
+    init(with restaurantProviding: RestaurantProviding, locationManager: CLLocationManager) {
+        manager = locationManager
+    }
     
     private var userLocation: CLLocation?
-    
-    private lazy var manager:  CLLocationManager = {
-        let manager = CLLocationManager()
-        manager.delegate = self
-        manager.requestWhenInUseAuthorization()
-        manager.requestLocation()
-        return manager
-    }()
-    
-    let regionRadius: CLLocationDistance = 4500
-        
-    private func createRegion(for location: CLLocation) -> MKCoordinateRegion {
-          let center = CLLocationCoordinate2D(latitude: location.coordinate.latitude,
-                                              longitude: location.coordinate.longitude)
-          
-          let region = MKCoordinateRegion(center: center,
-                                                    latitudinalMeters: regionRadius,
-                                                    longitudinalMeters: regionRadius)
-          
-          return region
-      }
+    private let manager:  CLLocationManager
     
     func requestUserLocation() {
         manager.requestLocation()
     }
 }
 
-extension MapViewModel: CLLocationManagerDelegate {
-    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        if let location = locations.first { // current user location
-            
-            let region = createRegion(for: location)
-            delegate?.shouldSetRegion(region)
-            
-   //         delegate?.didUpdateLocations(locations)
-            YelpAPI.fetchRestaurants(latitude: location.coordinate.latitude.description, longitude: location.coordinate.longitude.description) { result in
```

Comment:
the YelpAPI doesn't really belong here. I think we can do better by just reaching when the map changes